data_structure/TrieTree.cpp

```diff
@@ -1,97 +1,89 @@
+    nn -> isEndofWord = false;
     return nn;
 }
 
 // insert string into the trie
-void insert(trie *root, char *str)
-{
-    for (int i = 0; i < strlen(str); i++)
-    {
+void insert(trie * root, char * str) {
+    for (int i = 0; i < strlen(str); i++) {
         int j = str[i] - 'a';
-        if (root->arr[j])
-        {
-            root = root->arr[j];
-        }
-        else
-        {
-            root->arr[j] = createNode();
-            root = root->arr[j];
+        if (root -> arr[j]) {
+            root = root -> arr[j];
+        } else {
+            root -> arr[j] = createNode();
+            root = root -> arr[j];
         }
     }
-    root->isEndofWord = true;
+    root -> isEndofWord = true;
 }
 
 // search a string exists inside the trie
-bool search(trie *root, char *str, int index)
-{
-    if (index == strlen(str))
-    {
-        if (!root->isEndofWord)
+bool search(trie * root, char * str, int index) {
+    if (index == strlen(str)) {
+        if (!root -> isEndofWord)
             return false;
         return true;
     }
     int j = str[index] - 'a';
-    if (!root->arr[j])
+    if (!root -> arr[j])
         return false;
-    return search(root->arr[j], str, index + 1);
+    return search(root -> arr[j], str, index + 1);
 }
 /* removes the string if it is not a prefix of any  other 
  string, if it is then just sets the endofword to false, else 
  removes the given string*/
-bool deleteString(trie *root, char *str, int index)
-{
-    if (index == strlen(str))
-    {
-        if (!root->isEndofWord)
+bool deleteString(trie * root, char * str, int index) {
+    if (index == strlen(str)) {
+        if (!root -> isEndofWord)
             return false;
-        root->isEndofWord = false;
+        root -> isEndofWord = false;
         for (int i = 0; i < 26; i++)
             return false;
         return true;
     }
     int j = str[index] - 'a';
-    if (!root->arr[j])
+    if (!root -> arr[j])
         return false;
-    bool var = deleteString(root, str, index + 1);
-    if (var)
-    {
-        root->arr[j] = NULL;
-        if (root->isEndofWord)
+    bool
+    var = deleteString(root, str, index + 1);
+    if (var) {
+        root -> arr[j] = NULL;
+        if (root -> isEndofWord) {
             return false;
-        else
-        {
+        } else {
             int i;
             for (i = 0; i < 26; i++)
-                if (root->arr[i])
+                if (root -> arr[i])
                     return false;
             return true;
         }
     }
 }
 
-int main()
-{
-    trie *root = createNode();
-    insert(root, "hello");
-    insert(root, "world");
-    int a = search(root, "hello", 0);
-    int b = search(root, "word", 0);
+int main() {
+    trie * root = createNode();
+    insert(root, const_cast < char * > ("hello"));
+    insert(root, const_cast < char * > ("world"));
+    int a = search(root, const_cast < char * > ("hello"), 0);
+    int b = search(root, const_cast < char * > ("word"), 0);
```

Comment:
Would it make sense to use __string__ instead of __char *__ in these four instances?

Reply:
I think it's better to use **string**. But I didn't edit this algorithm. I just added const_cast < char * > to repair **warning: ISO C + + forbids converting a string constant to 'char *' [- wwrite strings]**  Perhaps the best way is to rewrite the algorithm with string.